### PR TITLE
option to disable classname prefixing

### PR DESCRIPTION
**Pull request by bicouy0**
This is an implementation of an option to remove the default behaviour of class prefixing.

* OrientGraphFactory.setLabelAsClassName() to disable the default prefixing of classnames with V_ or E_
* OrientGraph.labelToClassName() and OrientGraph.classNameToLabel() helpers
* OrientGraph.getEdgeIndexedKeys() and OrientGraph.getVertexIndexedKeys() verify the super class
* OrientGraph.createClass() throws if creating the same class with a different base class
* test suites modified

### CODE REVIEW

File: driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientElement.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public ORID id() {
     public String label() {
         String internalClassName = getRawDocument().getClassName();
         // User labels on edges/vertices are prepended with E_ or V_ . The user should not see that.
-        return internalClassName.length() == 1 ? INTERNAL_CLASSES_TO_TINKERPOP_CLASSES.get(internalClassName) : internalClassName.substring(2);
+        return internalClassName.length() == 1 ? INTERNAL_CLASSES_TO_TINKERPOP_CLASSES.get(internalClassName) : graph.classNameToLabel(internalClassName);
```
Comment (velo) [inline driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientElement.java]: Maybe ````graph.classNameToLabel()```` should handle this logic as well....

File: driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java [hunk 2]
```diff
@@ -166,6 +165,26 @@ public GraphComputer compute() throws IllegalArgumentException {
...
     }
 
+    /**
+     * Convert a label to orientdb class name
+     */
+    public String labelToClassName(String label, String prefix) {
+    	if (configuration.getBoolean(CONFIG_LABEL_AS_CLASSNAME, false)) {
+        	return label;
+    	}
+		return label.equals(prefix) ? prefix : prefix + "_" + label;
```
Comment (velo) [inline driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java]: @mpollmeier do you mind if I setup a standard format for the project?
https://github.com/velo/maven-formatter-plugin

Also, do you use eclipse? Do you have a prefer format for this project?

File: driver/src/test/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraphTest.java [hunk 3]
```diff
@@ -148,6 +149,26 @@ public void testPooledTransactionalGraph() throws Exception {
...
+        performBasicTests(graph);
+
+        Vertex vertex = graph.addVertex("VERTEX_LABEL");
+        assertEquals("VERTEX_LABEL", vertex.label());
+
+        try {
+            graph.addVertex("EDGE_LABEL");
+            Assert.fail("must throw unable to create different super class");
+        } catch (IllegalArgumentException e) {
+            // ok
```
Comment (velo) [inline driver/src/test/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraphTest.java]: would be nice to assert if the error message is matching what you expect

**Comment by velo** [comment]
LGTM

File: driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java (hunk 2 above)
Comment (mpollmeier) [inline driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java]: sure, please go ahead. I'm using intellij for this, but don't have hard feelings re formatting java code

**Comment by mpollmeier** [comment]
Can you explain the benefit of this please? The reason for prefixing the class names with V_ and E_ is so that you can use the same labels for vertices and edges. Why would anyone not like to do that?

File: driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java (hunk 2 above)
Comment (velo) [inline driver/src/main/java/org/apache/tinkerpop/gremlin/orientdb/OrientGraph.java]: ok, I will do my best to make it low impact!

**Comment by bicouy0** [comment]
The main reason is that I'm unable to use this driver with an existing databaase schema.

I already have a Person class in orientdb but hasLabel("Person") miss them.

Maybe i'm doing it wrong.

**Comment by mpollmeier** [comment]
ok yeah that makes sense. happy to merge after you've resolved the conflicts

**Comment by velo** [comment]
FWIW, I changed all master code format...
https://github.com/mpollmeier/orientdb-gremlin/blob/master/driver/src/config/eclipse/formatter/java.xml

Probably if you rebase should have little number of conflicts!

**Comment by velo** [comment]
I made the merge for you... you just need to apply to your branch and all shall start to look green
https://github.com/bicouy0/orientdb-gremlin/pull/1

**Comment by velo** [comment]
Seems travis is not happy about it.  I will merge once green.

**Pull request merged**